kube-runtime/src/utils/event_filter.rs

```diff
@@ -9,17 +9,17 @@ use pin_project::pin_project;
 #[pin_project]
 /// Stream returned by the [`applied_objects`](super::WatchStreamExt::applied_objects) and [`touched_objects`](super::WatchStreamExt::touched_objects) method.
 #[must_use = "streams do nothing unless polled"]
-pub struct EventFlatten<St> {
+pub struct EventFilter<St> {
```

Comment:
hm, yeah... possibly. maybe i'll hang on to this PR for a little bit first to see if it clashes with use cases for stream forwarding (since it was also running into the same issues.

Reply:
I have updated to `EventDecode` now.